### PR TITLE
Add Focus.parentNode to allow controlling the shape of the Focus tree.

### PR DESCRIPTION
**Pull request by gspencergoog**
## Description

This adds a new parameter to the `Focus` widget that allows the creator of the widget to specify the parent focus node that the focus widget should maintain, instead of using the widget hierarchy to look up the widget.

This allows users to create a focus tree that isn't the same shape as the widget tree, so that if they want to send focus to a widget in an overlay as though it were a child widget, and have the parent maintain focus, they can.

## Related Issues
 - https://github.com/flutter/flutter/issues/106923

## Tests
 - Added a test to make sure that the focus tree has the right shape and doesn't reparent to a different parent when a new Focus node is inserted into the tree.

### CODE REVIEW

File: packages/flutter/lib/src/widgets/focus_scope.dart [hunk 1]
```diff
@@ -153,6 +155,22 @@ class Focus extends StatefulWidget {
   // when then widget is updated.
   bool get _usingExternalFocus => false;
 
+  /// Returns the parent node to use when reparenting the [focusNode] for this
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/focus_scope.dart:158]: nit: maybe omit "Returns"? Most people will be reading this doc to figure out if they should set the parentNode property...

Comment (goderbauer) [reply packages/flutter/lib/src/widgets/focus_scope.dart:158]: Maybe say: "Optional parent node to use when..." (to really stress that most people don't want to set this)

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/widgets/focus_scope.dart (hunk 1 above)
Comment (gspencergoog) [reply packages/flutter/lib/src/widgets/focus_scope.dart:158]: Done.

**Pull request merged**